### PR TITLE
Use pombump to handle spark dep bumps.

### PR DESCRIPTION
**Pull request by vaikas**
Switch to using pombump. Has less CVEs, but they can not be currently removed. 
Trying to bump `libthrift-0.12.0.jar` to 13 or later does not compile because API has changed.
`jackson-mapper-asl-1.9.13` has not been updated since 2013.
`hadoop-client-runtime-3.3.6` is the latest
Others are shaded jars.

Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [ ] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [ ] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: spark/bump-packages.patch
Comment (jamonation) [inline spark/bump-packages.patch:1]: I think this patch file can disappear - I don't see it used anywhere now 🎉

File: spark.yaml [hunk 1]
```diff
@@ -48,7 +49,13 @@ pipeline:
     with:
       patches: make-distribution.patch bump-packages.patch
```
Comment (jamonation) [inline spark.yaml:50]:
```suggestion
      patches: make-distribution.patch
```

Same as comment below, I don't think? the bump-packages.patch gets used anywhere now with pombump doing the driving

**Review by jamonation — APPROVED** [review]

File: spark.yaml (hunk 1 above)
Comment (vaikas) [reply spark.yaml:50]: It does get used, there are some other kind of patching it does. For example:

```
+              <excludes>
+                <exclude>com.google.common.util.concurrent.internal.**</exclude>
+              </excludes>
```

Comment (vaikas) [reply spark.yaml:50]: We can add more capabilities into pombump as we go along.
But the above file also patches a `.scala` file.

**Review by vaikas — COMMENTED** [review]
There are still some other patches that need to get applied.

**Pull request merged**